src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/CustomCATest.java

```diff
@@ -169,6 +175,18 @@ private void givenCDAWithCustomCertificateAuthority(URI privateKeyUri, URI certi
         kernel.getContext().waitForPublishQueueToClear();
     }
 
+    /**
+     * Simulates a customer configuring a managed certificate authority.
+     */
+    private void givenManagedCertificateAuthorityConfiguration() throws ServiceLoadException {
```

Comment:
I think this is okay, but it's slightly misleading. To me, "given" implies a starting state. This is actually updating the state from whatever the "given" actually is.

I think what I'd prefer for us to do here is have the "givenX" preconditions established by starting nucleus with a given yaml configuration. These methods can then be used to show that the switch during a deployment works.

Thoughts?

Reply:
I think you are right this should be called just configureManagedCertificateAuthority I used to write tests a few years ago using a build that lazily ran the steps something like

```java
TestRunner.begin()
     .given(configureCustomCertificateAuthority("foo"))
     .when(serviceStarts())
     .then(assertThatXHappened())
     .run();
```

I won't do that know but I like that it highlights the steps and the focus of the test. I will change the name of the method though